stairway/src/main/java/bio/terra/stairway/Flight.java

```diff
@@ -97,13 +87,9 @@ protected void addStep(Step step, RetryRule retryRule) {
    */
   public void run() {
     try {
-      hookWrapper().startFlight(flightContext);
+      flightSupport.startFlightHook(flightContext);
```

Comment:
What if we made a client facing Flight class/interface and ported all of the code originally in this file to a `stairwayimpl/FlightImpl` class? I think that class could refer to `StairwayImpl` directly and we could get rid of `FlightSupport` entirely. As-is, FlightSupport seems like a strange bit of indirection where Flight still knows a lot about bits of the internal implementation, but doesn't get access to the impl directly.